aten/src/ATen/native/FractionalMaxPool2d.cpp

```diff
@@ -30,9 +30,15 @@ TORCH_META_FUNC(fractional_max_pool2d) (
   int64_t poolSizeW = pool_size[1];
 
   int64_t ndims = input.ndimension();
-  TORCH_CHECK(input.numel() > 0 && (ndims == 3 || ndims == 4),
-    "non-empty 3D or 4D (batch mode) tensor expected for input, but got: ",
-    ndims);
+  TORCH_CHECK(ndims == 3 || ndims == 4,
+              "fractional_max_pool2d(): Expected non-empty 3D or 4D tensor with optional 0-dim batch size, but got: ",
```

Comment:
"non-empty" isn't fully accurate now- why not just "Expected 3D or 4D input tensor, but got:" ?